packages/vscode-extension/src/extension.ts

```diff
@@ -46,13 +50,31 @@ export function activate(context: ExtensionContext) {
     clientOptions,
   );
 
+  // Initialize singletons
+  LangugageClientManager.instance = new LangugageClientManager(client);
+  PersistentConnectionManager.instance = new PersistentConnectionManager();
+  ConnectionRepository.instance = new ConnectionRepository(
+    context.globalState,
+    context.secrets,
+  );
+
+  // Ensure any connections are reset - set the connected flag to false
```

Comment:
actually, this makes no sense - we should honour a connection that's left as connected so our extension reconnects when it activates.. this mirrors existing functionality